src/AutoRest.CSharp/build/CodeGeneration.targets

```diff
@@ -17,14 +17,15 @@
     <AutoRestSharedCodeDirectory>$(MSBuildThisFileDirectory)../content/Generator.Shared/</AutoRestSharedCodeDirectory>
     <ManagementSharedCodeDirectory>$(MSBuildThisFileDirectory)../content/Management.Shared/</ManagementSharedCodeDirectory>
 
-    <_GenerateCode Condition="'$(AutoRestInput)' != ''">true</_GenerateCode>
+    <_GenerateCode Condition="'$(AutoRestInput)' != '' OR '$(CadlInput)' != ''">true</_GenerateCode>
     <UseDefaultNamespaceAndOutputFolder Condition="'$(UseDefaultNamespaceAndOutputFolder)' == ''">true</UseDefaultNamespaceAndOutputFolder>
     <_AutoRestCommand>node $(AutoRestEntryPoint) --max-memory-size=8192 --skip-csproj --skip-upgrade-check --version=$(AutoRestCoreVersion) $(AutoRestInput) $(AutoRestAdditionalParameters) --use=$(MSBuildThisFileDirectory)../tools/netcoreapp3.1/any/ --clear-output-folder=true --shared-source-folders=&quot;$(AzureCoreSharedCodeDirectory);$(AutoRestSharedCodeDirectory)&quot;</_AutoRestCommand>
     <_AutoRestCommand Condition="'$(UseDefaultNamespaceAndOutputFolder)' == 'true'">$(_AutoRestCommand) --output-folder=$(MSBuildProjectDirectory)/Generated --namespace=$(RootNamespace)</_AutoRestCommand>
+    <_CadlGenerateCommand Condition="'$(CadlInput)' != ''">npx cadl compile --emit @azure-tools/cadl-csharp --output-path $(MSBuildProjectDirectory)/Generated --option @azure-tools/cadl-csharp.clear-output-folder=true --option @azure-tools/cadl-csharp.save-inputs=false $(CadlInput)</_CadlGenerateCommand>
```

Comment:
We don't need `--option @azure-tools/cadl-csharp.save-inputs=false` right?

Reply:
NO. by default, cadl.json and Confifugration.json are intermediated files, and they may be used during debug, but they should not be part of the SDK library to ship.

Reply:
I mean save-input is by defualt false, why we pass it?

Reply:
got it, it is not needed. remove